File: src/main/java/workshop/part4/IDProvider.java
package workshop.part4;

import java.util.concurrent.atomic.AtomicInteger;

public final class IDProvider {

    private static final AtomicInteger counter = new AtomicInteger(0);

    public static String nextId() {
        return String.valueOf(getCounter());
    }

    private static Integer getCounter() {
        return counter.incrementAndGet();
    }


}

File: src/main/kotlin/workshop/part4/IDProvider.kt
package workshop.part4

import java.util.concurrent.atomic.AtomicInteger

object IDProvider {

    private val counter = AtomicInteger(0)

    fun nextId(): String {
        return getCounter().toString()
    }

    private fun getCounter(): Int {
        return counter.incrementAndGet()
    }


}
